src/database/memory.rs

```diff
@@ -493,7 +493,13 @@ macro_rules! populate_test_db {
         let input = if $is_coinbase {
             vec![$crate::bitcoin::TxIn::default()]
         } else {
-            vec![]
+            vec![$crate::bitcoin::TxIn {
+                previous_output: $crate::bitcoin::OutPoint {
+                    txid: Default::default(),
+                    vout: 5,
+                },
+                ..Default::default()
```

Comment:
Do we really need to hardcode transaction inputs in this way?? Seems little aggressive to me.. We are not using the transaction inputs from the `tx_meta` in any of the tests, so not an issue for now, but in future this can get awkward.. And whats the issue when you tried the #624 way?

Reply:
I am not really sure why this is so, but the trezor device doesn't behave properly if we leave that empty. When I test signing a psbt with trezor using code from #624, (and what is there on master), trezor (emulator) returns a broken psbt which cannot be deserialized.

Reply:
Ok I did some debugging and its failing because we actually don't set the transaction inputs in `tx_meta` in the `testutils` macro, unless its given in the macro call. 
https://github.com/bitcoindevkit/bdk/blob/c9b1b6d076cedcc389128d7d48d287aa2d1aa70d/src/testutils/mod.rs#L130-L133

And in `get_funded_wallet` we are calling `testutils` without any input. Because its a fake transaction, there's no input to specify.. So it made sense to have empty input for generic cases, but would fail where an input is expected. 

So if you add a hardcoded fake input in `get_funded_wallet` and then apply the #624 logic in `populate_test_db` it will work as expected.. I would suggest hard coding the input in `get_funded_wallet` instead of `populate_test_db`, as the later can be used in cases where we do know about transaction inputs..

Another option is to call `testutils` with `@coinbase true` in `get_funded_wallet`, so it will use the `TxIn::deafult()` value, which will also work..

Reply:
A new arguement in `get_funded_wallet()` for `@coinbase`. I think this would be the best (and easiest). With that I could skip the changes in `populate_test_db`.